spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/XsuaaServiceConfigurationCustom.java

```diff
@@ -72,4 +63,17 @@ public CredentialType getCredentialType() {
 		return credentials.getCredentialType();
 	}
 
+	@Override
+	public String getProperty(String name) {
+		return null;
+	}
+
+	@Override
+	public ClientIdentity getClientIdentity() {
+		ClientIdentity identity = new ClientCredentials(getClientId(), getClientSecret());
```

Comment:
I would reverse the logic -> first create ClientCertificate and check if it is valid and if not then create ClientCredentials as the X.509 becomes soon the new default auth type